src/diagonal.works/b6/api/functions/collections.go

```diff
@@ -341,3 +342,177 @@ var _ api.Collection = &flattenCollection{}
 func flatten(_ *api.Context, c api.Collection) (api.Collection, error) {
 	return &flattenCollection{c: c}, nil
 }
+
+type bound struct {
+	lower, upper interface{}
+	prettyPrint  interface{}
+	within       func(interface{}) (bool, error)
+}
+type buckets []bound
```

Comment:
flipping between buckets and bound now which is maybe a bit weird, but got attached to bound, lm keep it pls